plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/listener/QueryInsightsListener.java

```diff
@@ -69,11 +80,49 @@ public QueryInsightsListener(final ClusterService clusterService, final QueryIns
                 v -> this.queryInsightsService.getTopQueriesService(MetricType.LATENCY).setWindowSize(v),
                 v -> this.queryInsightsService.getTopQueriesService(MetricType.LATENCY).validateWindowSize(v)
             );
+        clusterService.getClusterSettings()
```

Comment:
nit: Could have a helper function to make the code cleaner since we are doing this for all MetricTypes.

Reply:
Good point, let me refactor this